### PR TITLE
created a dockerfile with liftoff

### PR DESCRIPTION
**Pull request by btkostner**
the start of strongback

#### Changes:
- created a `Dockerfile` to install liftoff

### CODE REVIEW

**Comment by btkostner** [comment]
Alrighty, this should be ready for a full review now. It has a workable dockerfile that can pull a repo, build it, and send info back to houston. This does not include the houston changes for strongback, as those will be in another PR. I will also break this up a bit later down the road to implement a hook system similar to flightcheck. This will allow cool things like dropping in a default apparmor profile if need be.

**Comment by btkostner** [comment]
Also the build is failing because of a linting error due to houston integration being partially done

File: houston/controller/hook/strongback.js [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+ */
+atc.on('build:finish', async (data) => {
+  const build = await Cycle.findOne({
+    
+  })
+
+  if (data.errors > 0) {
+    cycle.update({ '_status': 'FAIL' })
+  } else if (cycle.type === 'RELEASE') {
+    cycle.build()
```
Comment (fabianthoma) [inline houston/controller/hook/strongback.js]: Why are you triggering builds from the build:finish event?

File: houston/controller/hook/strongback.js [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    .then(() => cycle.update({ '_status': 'BUILD' }))
+  }
+
+  // TODO: update project information
+  const project = await cycle.getProject()
+  for (const i in data.issues) {
+    project.postIssue(data.issues[i])
+  }
```
Comment (fabianthoma) [inline houston/controller/hook/strongback.js]: Are issues the only thing we need to update? What about released version etc. ?

File: strongback/docker/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,43 @@
+# Houston docker file
+# Builds an ubuntu base with liftoff for packaging debian applications
```
Comment (fabianthoma) [inline strongback/docker/Dockerfile]: Why exactly are we doing this? liftoff creates it's own chroot, so the filesystem is isolated, running it on it's own secured VM should be good enough once we let other people into houston

**Comment by fabianthoma** [comment]
Yeah adding binary files isn't really what we should do: `strongback/docker/liftoff_0.1_amd64.deb`

**Comment by btkostner** [comment]
Because liftoff is currently in a private repo with no available ppa or build

File: houston/controller/hook/strongback.js (hunk 2 above)
Comment (btkostner) [inline houston/controller/hook/strongback.js]: released version is usually calculated by the latest release that has a 'FINISH' status. As far as other things, there is some more stuff to this hook in 'strongback-aptly' branch.

File: strongback/docker/Dockerfile (hunk 3 above)
Comment (btkostner) [inline strongback/docker/Dockerfile]: It seemed like a good idea at the time, as extra protection and isolation. This means that strongback would have to be ran on a debian based system (which shouldn't be a problem now, but I'm not a future teller). If you believe liftoff is good enough by itself I can remove the docker dependency.

We will also need the server to have liftoff installed with the houston user having sudo access to that command.

File: houston/controller/hook/strongback.js (hunk 1 above)
Comment (btkostner) [inline houston/controller/hook/strongback.js]: I am not sure. It looks like it was committed half way done. Will fix.

File: strongback/docker/Dockerfile (hunk 3 above)
Comment (btkostner) [inline strongback/docker/Dockerfile]: Also worth noting that docker will probably be involved down the line with automated testing.

File: houston/controller/hook/strongback.js (hunk 1 above)
Comment (fabianthoma) [inline houston/controller/hook/strongback.js]: Will fix down the line or in this PR? Cause your PRs all depend on each other, which could make changing this now weird

File: strongback/docker/Dockerfile (hunk 3 above)
Comment (fabianthoma) [inline strongback/docker/Dockerfile]: Alright we can change this down the line if it causes us trouble, right now it's working fine, although there probably is a slight performance drop here.

**Pull request merged**